### PR TITLE
Passcode UI updates

### PR DESCRIPTION
**Pull request by brandonpage**
- Fix UI not changing after first unlock on app upgrade
- Fix keyboard focus
- Don't allow passcode length to be lowered
- Fix passcode does not match reentering when creating

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java [hunk 1]
```diff
@@ -616,12 +616,14 @@ public void setMinPasscodeLength(Context ctx, int minPasscodeLength) {
...
     public void setPasscodeLength(Context ctx, int passcodeLength) {
     	if (passcodeLength > this.passcodeLength) {
-            if (hasStoredPasscode(ctx)) {
+            if (hasStoredPasscode(ctx) && passcodeLengthKnown) {
                 this.passcodeChangeRequired = true;
             }
+
+            this.passcodeLength = passcodeLength;
     	}
-        this.passcodeLength = passcodeLength;
```
Comment (brandonpage) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java]: @bhariharan Why was passcode length requirement allowed to be lowered?

Comment (bhariharan) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java]: Passcode length could be lowered if you log out of an account with a longer length requirement, when there are 2 accounts with passcode enabled.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java [hunk 2]
```diff
@@ -642,4 +647,11 @@ private void clearUi() {
         biometricBox.setVisibility(View.GONE);
         fingerImage.setVisibility(View.GONE);
     }
+
+    private void hideKeyboard() {
+        InputMethodManager imm = (InputMethodManager) this.getSystemService(Activity.INPUT_METHOD_SERVICE);
+        if (this.passcodeField != null) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: `this` is implicit. You don't need to explicitly add it.

Comment (brandonpage) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: Whoops.  Made the change locally, I will commit it when UX feedback comes back.

**Pull request merged**